test/stdlib/StringIndex.swift

```diff
@@ -1086,3 +1086,90 @@ suite.test("Substring.removeSubrange entire range")
   expectTrue(b.isEmpty)
 #endif
 }
+
+if #available(SwiftStdlib 5.8, *) {
```

Comment:
Minor quibble: ideally skipped tests should be visible as having been skipped in the output.

Reply:
Agreed. Sadly we don't have a good way to express that in the current language -- the only thing we could do is to add the availability check as a skip condition while also repeating it within the test body, which seems unreasonably complicated. 😞